packages/flutter_tools/lib/src/web/web_device.dart

```diff
@@ -131,7 +131,13 @@ class ChromeDevice extends Device {
   }) async {
     // See [ResidentWebRunner.run] in flutter_tools/lib/src/resident_web_runner.dart
     // for the web initialization and server logic.
-    _chrome = await chromeLauncher.launch(platformArgs['uri']);
+    final String url = platformArgs['uri'];
+    if (debuggingOptions.browserLaunch) {
+      _chrome = await chromeLauncher.launch(url);
+    } else {
+      printStatus('Waiting for connection from Dart debug extension at $url', emphasis: true);
+      logger.sendNotification(url, progressId: 'debugExtension');
```

Comment:
This doesn't seem to work - Daemon seems to use NotifyingLogger which silently swallows sendNotification. I think maybe it needs to be sent using `startProgress`?